vms/platformvm/vm.go

```diff
@@ -480,3 +481,21 @@ func (vm *VM) VerifyHeightIndex(_ context.Context) error {
 func (vm *VM) GetBlockIDAtHeight(_ context.Context, height uint64) (ids.ID, error) {
 	return vm.state.GetBlockIDAtHeight(height)
 }
+
+// We need to grab the context lock here to avoid racy behavior with
+// transaction verification + mempool modifications.
+//
```

Comment:
This comment doesn't really make a ton of sense (because the lock isn't grabbed here). Maybe we should comment that `Network.IssueTx` grabs the lock instead?

Reply:
I'll just delete this comment since it's already documented in `network`.